### ISSUE TITLE
KAL skips List tags linting

### ISSUE THREAD

**Issue by avalanche123** [kubernetes-sigs/kube-api-linter#147]
Looks like #134 introduced a change that breaks `jsontags` linter for list types.

`jsontags` [relies on `inspect.InspectFields`](https://github.com/kubernetes-sigs/kube-api-linter/blob/main/pkg/analysis/jsontags/analyzer.go#L74-L76)
[`InspectFields` will skip fields in list types](https://github.com/kubernetes-sigs/kube-api-linter/blob/main/pkg/analysis/helpers/inspector/inspector.go#L99-L102)

This makes it possible to define a list type like:

```go
// +kubebuilder:object:root=true

// ResourceList contains a list of Resource
type ResourceList struct {
	metav1.TypeMeta `         json:",inline"`
	metav1.ListMeta `         json:"metadata"`
	Items           []Resource
}
```

this will lead to json serialization of list as:

```json
{
  "apiVersion": "group/version"
  "kind": "ResourceList",
  "Items": []
}
```

and such list will miss its items when deserialized back using client-go.

**Comment by JoelSpeed** [comment]
I suspect it was actually https://github.com/kubernetes-sigs/kube-api-linter/pull/75 that introduced this issue no? Or have you tried with a version including #75 but without #134? 

In general, we don't really care about the List type fields and providing linting feedback for them, but you're right, this would be an issue so we can/should report this. We may need to just ignore the helper functions for iterating over fields when implementing the jsontags linter

Any time to raise a PR suggesting an appropriate fix?
